First pass at license enforcement update (#2512)
FlowFuse/website: src/changelog/2024/08/enterprise-license-update.md

---
title: Enforcing Enterprise Restrictions
description: "A stricter approach to expired licenses"

date: 2024-07-24 13:00:00.0

authors: ["ben-hardill"]
tags:
    - license
    - enterprise
---

Starting with the v2.8.0 release we will be taking a slightly stricter 
approach to Self Hosted Licenses. Previously when a license expired the
platform would continue to run with the same capabilities as when the
license was valid.

> If you are currently running with older version and an expired license

Author: ```suggestion
If you are currently running with older version with  an expired license
```

> If you are currently running with older version with  an expired license

Author: ```suggestion
If you are currently running with older version with an expired license
```

> If you are currently running with older version with an expired license

Reviewer: ```suggestion
If you are currently running with an older version with an expired license
```

> Starting with the v2.8.0 release we will be taking a slightly stricter 
> approach to Self Hosted Licenses. Previously when a license expired the
> platform would continue to run with the same capabilities as when the
> license was valid.
> 
> If you are currently running with older version with an expired license
> please contact the Sales team to discuss renewing your license before 
> upgrading. You can check your current license on the 
> Admin Settings -> Overview page.

Reviewer 2: I'm not a fan of the royal we: (hope I kept the 80 char limit)

```suggestion
Starting with the v2.8.0 release FlowFuse will be taking a slightly stricter 
approach to License enforcement outside of FlowFuse Cloud. Previously
when a license expired the platform would continue to run with the same
capabilities as when the license was valid.

If you are currently running with older version with an expired license
please [contact the Sales team](/contact-us/) to discuss renewing your
license before upgrading. You can check your current license on the 
Admin Settings -> Overview page.
```

> 

Reviewer 2: Do changelogs allow for a "fold"? Do we need a `<!--more-->` here?

Author: Changelogs don't normally have a fold from the previous examples

## Notifications

Starting 30 days before the license expires:

- Admin users will see a banner showing the number of days left.

   ![expiring banner](images/expiring-banner.png)
- A email will be sent to all Admin users.

The email will be sent again every Sunday evening in the 30 days.

## After Expiry

- All running instances will be suspended.

> - Instances will not be stared.

Reviewer 2: ```suggestion
- Instances will not be started.
```

> - No new Instances or Devices can be created.
> 
>    ![expired banner](images/expired-banner.png)
> - A email will be sent to all Admin users saying the license has expired.
> - A banner showing the license as expired will be shown to all users.

Author: ```suggestion
- No new Instances or Devices can be created.
- A email will be sent to all Admin users saying the license has expired.
- A banner showing the license as expired will be shown to all users.

   ![expired banner](images/expired-banner.png)
```

- All connections to the MQTT broker will be refused (This will impact 
Project Nodes and Device control).

## Applying a New License

A new license can be applied in the Admin Settings -> Settings section.

> Once applied you will be able to restart all existing Instances and 

Reviewer 2: When applying a license, should the application be restarted? https://flowfuse.com/docs/upgrade/open-source-to-premium/ suggests it wouldn't

Author: We don't have automatic restart this time round (we also don't have a flag for it in the db). So users will have to manually restart instances.

Reviewer 2: I wasn't clear, I'm sorry. The app server, fuse app, doesn't need to be restarted?

Author: A restart is only needed for the first time a license is applied, if the license has just expired updating it via the web interface is sufficient as all the feature flags are already loaded.
